CmrUI/Screens/News Feed tab/NewsFeedVC.swift

```diff
@@ -0,0 +1,333 @@
+            feedDetailedVC.feedItem = item
+            //feedDetailedVC.transitioningDelegate = self
+            feedDetailedVC.modalPresentationStyle = UIModalPresentationStyle.custom
+            feedDetailedVC.navigationItem.hidesBackButton = true
+        }
+    }
+
+    // MARK: UICollectionViewDataSource
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 3
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return dataSource.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if dataSource[indexPath.item].feedType == FeedType.feedImage {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants().kReuseCellID, for: indexPath) as! FeedNewsCell
+            
+            cell.backgroundImageView.image = dataSource[indexPath.item].image
+            cell.titleTextView.text = dataSource[indexPath.item].title
+            cell.subtitleLabel.text = dataSource[indexPath.item].subtitle
+            cell.backgroundColor = UIColor.red
+            cell.isUserInteractionEnabled = true
+            cell.titleTextView.isUserInteractionEnabled = false
+            return cell
+        }
+        else {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants().kReuseCellTextID, for: indexPath) as! FeedNewsTextCell
+            cell.titleTextView.text = dataSource[indexPath.item].title
+            cell.subtitleLabel.text = dataSource[indexPath.item].subtitle
+            cell.isUserInteractionEnabled = true
+            cell.titleTextView.isUserInteractionEnabled = false
+            return cell
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplaySupplementaryView view: UICollectionReusableView, forElementKind elementKind: String, at indexPath: IndexPath) {
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        if let supplementaryView = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionElementKindSectionHeader, withReuseIdentifier: Constants().kReuseHeader, for: indexPath) as? FeedNewsSectionHeader {
+            // Configure Supplementary View
+            supplementaryView.backgroundColor = .random()
+            supplementaryView.titleHeader.text = "WEDNESDAY \(indexPath.section + 1) JUNE "
+            
+            return supplementaryView
+        }
+        else {
+            return UICollectionReusableView()
+        }
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        transitionIndexPath = indexPath
+        
+        self.performSegue(withIdentifier: "DetailedFeedNews", sender: nil)
```

Comment:
@vmsavka Better use the SegueID enum.

Reply:
agree with you
We have a lot of constants for predefined colors, fonts, image names, sizes. In my mind it would be useful to apply SwiftGen tool. What do you think about it?

Reply:
As long as we stick to it, sure. I don't have any experience with it, but it looks like a useful tool for these purposes.